packages/twenty-front/src/modules/context-store/hooks/useContextStoreSelectedRecords.ts

```diff
@@ -0,0 +1,109 @@
+import { contextStoreCurrentObjectMetadataIdState } from '@/context-store/states/contextStoreCurrentObjectMetadataIdState';
+import { contextStoreTargetedRecordsFiltersState } from '@/context-store/states/contextStoreTargetedRecordsFilters';
+import { contextStoreTargetedRecordsState } from '@/context-store/states/contextStoreTargetedRecordsState';
+import { useObjectMetadataItemById } from '@/object-metadata/hooks/useObjectMetadataItemById';
+import { RecordGqlFields } from '@/object-record/graphql/types/RecordGqlFields';
+import { useFindManyRecords } from '@/object-record/hooks/useFindManyRecords';
+import { turnFiltersIntoQueryFilter } from '@/object-record/record-filter/utils/turnFiltersIntoQueryFilter';
+import { useFindManyParams } from '@/object-record/record-index/hooks/useLoadRecordIndexTable';
+import { makeAndFilterVariables } from '@/object-record/utils/makeAndFilterVariables';
+import { useRecoilValue } from 'recoil';
+
+const getFilterForSelectedRecords = (
+  selectedRecordIds: string[] | 'all',
+  excludedRecordIds: string[],
+  queryFilter: any,
+) => {
+  if (selectedRecordIds === 'all') {
+    if (excludedRecordIds.length > 0) {
+      return makeAndFilterVariables([
+        queryFilter,
+        {
+          not: {
+            id: {
+              in: excludedRecordIds,
+            },
+          },
+        },
+      ]);
+    }
+    return queryFilter;
+  }
+
+  return makeAndFilterVariables([
+    queryFilter,
+    {
+      id: {
+        in: selectedRecordIds,
+      },
+    },
+  ]);
+};
+
+export const useContextStoreSelectedRecords = ({
+  limit = undefined,
+  recordGqlFields,
+}: {
+  limit?: number;
+  recordGqlFields?: RecordGqlFields;
+}) => {
+  const contextStoreTargetedRecords = useRecoilValue(
+    contextStoreTargetedRecordsState,
+  );
+
+  const contextStoreCurrentObjectMetadataId = useRecoilValue(
+    contextStoreCurrentObjectMetadataIdState,
+  );
+
+  const contextStoreTargetedRecordsFilters = useRecoilValue(
+    contextStoreTargetedRecordsFiltersState,
+  );
+
+  const { objectMetadataItem } = useObjectMetadataItemById({
```

Comment:
objectMetadataItem could be undefined, I can see you handle it below in multiple places but don't want want to throw early in this case? Does it make sense to run that code without it?

Reply:
+1 we should throw in the useObjectMetadataItemById